server/repository/src/db_diesel/item_store_join.rs

```diff
@@ -0,0 +1,147 @@
+use diesel::prelude::*;
+
+use crate::{RepositoryError, StorageConnection, Upsert};
+
+table! {
+  item_store_join (id) {
+    id -> Text,
+    item_link_id  -> Text,
+    store_id -> Text,
+    default_sell_price_per_pack -> Double,
+  }
+
+}
+
+#[derive(Clone, Queryable, Insertable, AsChangeset, Debug, PartialEq, Default)]
+#[diesel(table_name = item_store_join)]
+pub struct ItemStoreJoinRow {
+    pub id: String,
+    pub item_link_id: String,
+    pub store_id: String,
+    pub default_sell_price_per_pack: f64,
+}
+
+pub struct ItemStoreJoinRowRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+pub trait ItemStoreJoinRowRepositoryTrait<'a> {
+    fn find_one_by_id(&self, row_id: &str) -> Result<Option<ItemStoreJoinRow>, RepositoryError>;
+    fn find_one_by_item_and_store_id(
+        &self,
+        item_link_id: &str,
+        store_id: &str,
+    ) -> Result<Option<ItemStoreJoinRow>, RepositoryError>;
+    fn upsert_one(&self, row: &ItemStoreJoinRow) -> Result<(), RepositoryError>;
+}
+
+impl<'a> ItemStoreJoinRowRepositoryTrait<'a> for ItemStoreJoinRowRepository<'a> {
+    fn find_one_by_id(&self, row_id: &str) -> Result<Option<ItemStoreJoinRow>, RepositoryError> {
+        use self::item_store_join::dsl::*;
```

Comment:
For fun added a usage of dsl in the manner that Diesel intended. Falls over if there are conflicts, such as in `find_one_by_item_and_store_id` as it's parameter names are the same as the dsl exports field names of the table